src/lib/define-dynamic-block.js

```diff
@@ -21,17 +21,49 @@ const setupCustomContextMenu = (ScratchBlocks, contextMenuInfo, extendedOpcode)
                     enabled: true,
                     text: contextOption.text,
                     callback: () => {
+                        const blockInfo = JSON.parse(this.blockInfoText);
+                        const contextOptionCallbackData = {
+                            blockInfo: blockInfo,
+                            blockId: this.id
+                        };
                         if (contextOption.builtInCallback) {
                             switch (contextOption.builtInCallback) {
                             case 'EDIT_A_PROCEDURE':
                                 // TODO FILL THIS IN
+                                // E.g. make use of guiContext here to bring up
+                                // the edit custom proc modal
                                 break;
-                            case 'RENAME_A_VARIABLE':
-                                // TODO FILL THIS IN
+                            case 'RENAME_A_VARIABLE': {
+                                // Extract variable info from this block
+                                const varName = blockInfo.text;
+                                // TODO it's not great that the following is hard coded...
+                                // We could use ScratchBlocks.SCALAR_VARIABLE_TYPE and
+                                // ScratchBlocks.LIST_VARIABLE_TYPE here instead, but
+                                // given that we eventually want to decouple this code from
+                                // using ScratchBlocks variables, we ultimately want these
+                                // strings to come from scratch-vm instead
+                                const varType = blockInfo.opcode === 'variable' ? '' : 'list';
```

Comment:
Yep - I agree with your comment. I have yet to hear about a good way to do this in JS though it could be added to `scratch-vm`'s main export list. That's what I've been doing to access `scratch-gui` internals from `scratch-desktop`, though I've been trying to minimize that.